### ISSUE TITLE
[ENH] weighted average of distances & distance composites

### ISSUE THREAD

**Issue by fkiraly** [sktime/sktime#1888]
There should be a wrapper class `CompositeDistance` (or similar) that allows to specify, as parameters:

* an iterable of distances, or just a single distance
* a way to aggregate, e.g., `"mean"` or `"median"` (string) or a `numpy` function
* an iterable of weights (optional)
* an iterable of variables, or lists of variables (optional)

If iterables are passed, they must be of same length.

The distance computed is the aggregate/average of: the i-th distance (the one distance, if not an iterable), computed on the i-th variable (set; all variables if none provided), multiplied by the i-th weight (1 if not provided).

**Comment by abheesht17** [comment]
Hello, @fkiraly! I would like to take this up.

I'll describe what you've said above, and you can tell me when I've understood it correctly. 

I'll be given a couple of dataframes. For example,

```
>>> a
     x1    x2    x3
0  0.34  0.87  0.30
1  0.45  0.56  0.67
2  0.12  0.32  0.31
>>> b
     x1    x2    x3
0  0.98  0.79  0.03
1  0.56  0.69  0.76
2  0.25  0.25  0.12
```

For distances, I'll be given something like "SquaredDistance" and "EuclideanDistance".

Assume
```
distances = ["SquaredDistance", "EuclideanDistance"]
weights = [0.4, 0.6]
variables = [("x1", "x2"), "x3"]
```

Essentially, this is what I'll have to compute:

```
composite_distance = 0.4 * squared_distance(a["x1", "x2"], b["x1", "x2"]) + 0.6 * euclidean_distance(a["x3"], b["x3"]).
```

Is this interpretation correct?

**Comment by fkiraly** [comment]
Yes, or almost, depending on what you were thinking.
The last line would eventually be computed, but as follows.

First, you write sth
`composite_distance = CompositeDistance(distances= [SquaredDistance, EuclideanDistance], weights = [0.4, 0.6], variables=[["x1", "x2"], "x3")`

and then you can call

`composite_distance(a, b)`, this computes the value in the last line of yours.

**Comment by fkiraly** [comment]
The extension template to follow would (probably) be this:
https://github.com/alan-turing-institute/sktime/blob/main/extension_templates/dist_kern_panel.py

**Comment by abheesht17** [comment]
Great! Sounds good. Will open a PR soon :)

**Comment by fkiraly** [comment]
implemented as `CombinedDistance` in #3264, forgot to close this.
